pkg/progress/supermin.go

```diff
@@ -0,0 +1,338 @@
+package progress
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/osbuild/image-builder-cli/pkg/util"
+)
+
+var superminInitScriptFmt = `#!/bin/sh
+# inspired by https://github.com/coreos/coreos-assembler/blob/main/src/supermin-init-prelude.sh
+# we need less because osbuild does most of its work via buildroots
+
+set -e
+
+export PATH=/usr/sbin:/usr/bin:/sbin:/bin
+
+mount -t proc /proc /proc
+mount -t sysfs /sys /sys
+mount -t cgroup2 cgroup2 -o rw,nosuid,nodev,noexec,relatime,seclabel,nsdelegate,memory_recursiveprot /sys/fs/cgroup
+mount -t devtmpfs devtmpfs /dev
+
+# auto-reboot on kernel panic (e.g. if anything in this script dies)
+echo 1 > /proc/sys/kernel/panic
+
+# this is also normally set up by systemd in early boot
+ln -s /proc/self/fd/0 /dev/stdin
+ln -s /proc/self/fd/1 /dev/stdout
+ln -s /proc/self/fd/2 /dev/stderr
+
+# need /dev/shm for podman
+mkdir -p /dev/shm
+mount -t tmpfs tmpfs /dev/shm
+
+# osbuild needs /run
+mkdir -p /run
+mount -t tmpfs tmpfs /run
+
+### from here on we diverge from the above core-assembler script
+
+# ensure network
+/usr/sbin/dhclient eth0
+
+# central for osbuild
+mknod /dev/loop-control c 10 237
+
+# map in the host /output dir, virtiofsd support uid remapping so this
+# should be fine
+mkdir -p /output
+mount -t virtiofs osbuild_output /output
+
+# We cannot put the host osbuild "store" on our /store dir even with
+# virtiofsd as there will be issues with uid remapping and selinux
+# labels.
+mkdir -p /host/store
+mount -t virtiofs osbuild_store /host/store
+# rsync/cp the caches so that our build is faster
+echo "Populate /store from host"
+mkdir /store
+rsync -a --exclude=./tmp /host/store/ /store
+
+# fetch/cache sources first so that the host cache gets updated
+echo "Fetching sources"
+osbuild \
+  --cache /store \
+  /output/manifest.json
+# rsync
+rsync -a --exclude=./tmp /store/ /host/store
+
+echo "Running osbuild"
+osbuild \
+  --export %s \
+  --output-directory /output \
+  --cache /store \
+  /output/manifest.json
```

Comment:
Not entirely happy since we've been doing a bit of effort to centralize how we format calling `osbuild`. But this adds a way of calling it that exists entirely outside of that approach. Don't really know a better way either.

Reply:
Yeah, we need a new "osbuild/exec" or something module that is just passed to the various places to have only a single implementation of executing osbuild. Its on my list for a while, maybe now is a good time to do it.

Reply:
That doesn't need to be in this PR, as long as we track (somewhere) that we'll unify it :)